packages/@aws-cdk/aws-appconfig-alpha/lib/configuration.ts

```diff
@@ -0,0 +1,881 @@
+
+  /**
+   * The latest version number of the hosted configuration.
+   */
+  public readonly latestVersionNumber?: number;
+
+  /**
+   * The version label of the hosted configuration.
+   */
+  public readonly versionLabel?: string;
+
+  /**
+   * The version number of the hosted configuration.
+   */
+  public readonly versionNumber?: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the hosted configuration version.
+   */
+  public readonly hostedConfigurationVersionArn: string;
+
+  /**
+   * The ID of the configuration profile.
+   */
+  public readonly configurationProfileId: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the configuration profile.
+   */
+  public readonly configurationProfileArn: string;
+
+  private readonly _cfnConfigurationProfile: CfnConfigurationProfile;
+  private readonly _cfnHostedConfigurationVersion: CfnHostedConfigurationVersion;
+
+  constructor(scope: Construct, id: string, props: HostedConfigurationProps) {
+    super(scope, id, props);
+
+    this._cfnConfigurationProfile = new CfnConfigurationProfile(this, 'ConfigurationProfile', {
+      applicationId: this.applicationId,
+      locationUri: 'hosted',
+      name: this.name!,
+      description: this.description,
+      type: this.type,
+      validators: this.validators,
+    });
+    this.configurationProfileId = this._cfnConfigurationProfile.ref;
+    this.configurationProfileArn = Stack.of(this).formatArn({
+      service: 'appconfig',
+      resource: 'application',
+      resourceName: `${this.applicationId}/configurationprofile/${this.configurationProfileId}`,
+    });
+    this.extensible = new ExtensibleBase(scope, this.configurationProfileArn);
+
+    this.content = props.content.content;
+    this.contentType = props.contentType || 'application/json';
+    this.latestVersionNumber = props.latestVersionNumber;
+    this.versionLabel = props.versionLabel;
+    this._cfnHostedConfigurationVersion = new CfnHostedConfigurationVersion(this, 'Resource', {
+      applicationId: this.applicationId,
+      configurationProfileId: this.configurationProfileId,
+      content: this.content,
+      contentType: this.contentType,
+      description: this.description,
+      latestVersionNumber: this.latestVersionNumber,
+      versionLabel: this.versionLabel,
+    });
+    this._cfnHostedConfigurationVersion.applyRemovalPolicy(RemovalPolicy.RETAIN);
+
+    this.versionNumber = this._cfnHostedConfigurationVersion.ref;
+    this.hostedConfigurationVersionArn = Stack.of(this).formatArn({
+      service: 'appconfig',
+      resource: 'application',
+      resourceName: `${this.applicationId}/configurationprofile/${this.configurationProfileId}/hostedconfigurationversion/${this.versionNumber}`,
+    });
+
+    this.addExistingEnvironmentsToApplication();
+    this.deployConfigToEnvironments();
+  }
+
+  protected getDeploymentHash(environment: IEnvironment): string {
+    const combinedString = `
+      ${this.application!.name!}
```

Comment:
I added the location type to this deployment hash for sourced configuration but we don't want to add any other properties since we only want to trigger a new deployment if the configuration content changes (any other changes made will be picked up when the config changes)